src/core/cgroup.c

```diff
@@ -777,6 +777,32 @@ void cgroup_oomd_xattr_apply(Unit *u, const char *cgroup_path) {
                 unit_remove_xattr_graceful(u, cgroup_path, "user.oomd_omit");
 }
 
+void cgroup_log_xattr_apply(Unit *u, const char *cgroup_path) {
+        ExecContext *c;
+
+        assert(u);
+
+        c = unit_get_exec_context(u);
+        if (!c)
+                return;
+
+        if (c->log_filter_regex) {
+                unit_set_xattr_graceful(u, cgroup_path,
+                                        "user.journald_log_filter_regex",
+                                        c->log_filter_regex,
+                                        strlen(c->log_filter_regex));
+                unit_set_xattr_graceful(u, cgroup_path,
+                                        "user.journald_log_filter_allow_list",
+                                        c->log_filter_allow_list ? "1" : "0",
+                                        1);
```

Comment:
I wonder if it'd be simpler to use just one cgroup xattr that includes the "\~" and only store the regex (including the "\~") in pid1 instead of two different properties?